### ISSUE TITLE
KubernetesClient Changed ListNamespacedPodAsync()

### ISSUE THREAD

**Issue by henric-root** [dotnet/orleans#8564]
We ran into this error when adding the KubernetesClient NuGet package (11.0.44) to one of our projects:
```
*** Unhandled exception: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<k8s.Models.V1PodList> k8s.CoreV1OperationsExtensions.ListNamespacedPodAsync(k8s.ICoreV1Operations, System.String, System.Nullable`1<Boolean>, System.String, System.String, System.String, System.Nullable`1<Int32>, System.String, System.String, System.Nullable`1<Int32>, System.Nullable`1<Boolean>, System.Nullable`1<Boolean>, System.Threading.CancellationToken)'.
         at Orleans.Hosting.Kubernetes.KubernetesClusterAgent.OnStart(CancellationToken cancellation)
         at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
         at Orleans.Hosting.Kubernetes.KubernetesClusterAgent.OnStart(CancellationToken cancellation)
         at Orleans.LifecycleExtensions.Observer.OnStart(CancellationToken ct) in /_/src/Orleans.Core.Abstractions/Lifecycle/LifecycleExtensions.cs:line 123
         at Orleans.Runtime.SiloLifecycleSubject.MonitoredObserver.OnStart(CancellationToken ct) in /_/src/Orleans.Runtime/Lifecycle/SiloLifecycleSubject.cs:line 134
         at Orleans.LifecycleSubject.OnStart(CancellationToken cancellationToken) in /_/src/Orleans.Core/Lifecycle/LifecycleSubject.cs:line 118
         at Orleans.Runtime.Scheduler.AsyncClosureWorkItem.Execute() in /_/src/Orleans.Runtime/Scheduler/ClosureWorkItem.cs:line 34
         at Orleans.Runtime.Silo.StartAsync(CancellationToken cancellationToken) in /_/src/Orleans.Runtime/Silo/Silo.cs:line 213
         at Orleans.Hosting.SiloHostedService.StartAsync(CancellationToken cancellationToken) in /_/src/Orleans.Runtime/Hosting/SiloHostedService.cs:line 28
         at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
         at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
         at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
         at Program.<Main>$(String[] args) in C:\Users\[snip...]:line 32
         at Program.<Main>(String[] args)
```
It looks like KubernetesClient has changed the arguments to `.ListNamespacedPodAsync()` sometime between the KubernetesClient version 10.0.31 pulled in when referencing Microsoft.Orleans.Hosting.Kubernetes 7.2.1, and the current version 11.0.44 of KubernetesClient.

We switched to KubernetsClient 10.0.31 to avoid this problem.  Should later version of KubernetesClient 10.x.x also work?  We have another NuGet package, Orleans.Clustering.Kubernetes, that also depends on KubernetesClient but so far this hasn't been a problem since it also seems happy with 10.0.31.  This all seems a little fragile and we're concerned that this might lead to more problems as one or more of these packages are updated (particularly if such updates include security issues).

**Comment by jfuksa** [comment]
Hi,
the problem seems to be the use of the KubernetesClient library version 11.0.x.

Orleans itself in version 7.2.1 has the lowest required version of this library 10.0.16.

If I also add a reference to:
`<PackageReference Include="AspNetCore.HealthChecks.UI" Version="7.0.2" />`
it transiently enforces a dependency on KubernetesClient ≥ 11.0.44 and this problem occurs.

**Comment by ReubenBond** [comment]
This is not the first time the KubernetesClient API has been broken. If you are willing, please submit a PR to update the [KubernetesClient version](https://github.com/dotnet/orleans/blob/075df202679e8e620f698cede7e97c52ec639b20/Directory.Packages.props#L63)

**Comment by henric-root** [comment]
> This is not the first time the KubernetesClient API has been broken. If you are willing, please submit a PR to update the [KubernetesClient version](https://github.com/dotnet/orleans/blob/075df202679e8e620f698cede7e97c52ec639b20/Directory.Packages.props#L63)

https://github.com/dotnet/orleans/pull/8576

**Comment by henric-root** [comment]
If this happens every now and then, perhaps this should be brought to the attention of the KubernetesClient folks?

Also, the NuGet UI for the solution offered to update 45 other packages and warned that one was deprecated (Microsoft.Extensions.Configuration.AzureKeyVault).

**Comment by cdemi** [comment]
Can I suggest that OrleansContrib/Orleans.Clustering.Kubernetes be adopted by the Orleans team and in the Orleans repo so that version between Orelans Clustering and Orleans Hosting are always in sync?

I think most Orleans Deployments use both the Clustering and Hosting components so I think this would streamline it

**Comment by ReubenBond** [comment]
> Can I suggest that OrleansContrib/Orleans.Clustering.Kubernetes be adopted by the Orleans team and in the Orleans repo so that version between Orelans Clustering and Orleans Hosting are always in sync?

> I think most Orleans Deployments use both the Clustering and Hosting components so I think this would streamline it

I'm open to it. It should be a new issue or ideally a PR. There are likely other Kubernetes integrations which we could have, too.

Closing this issue since it was fixed in #8576
